### ISSUE TITLE
Inaccuracy: High Contrast Mode

### ISSUE THREAD

**Issue by JAWS-test** [a11yproject/a11yproject.com#882]
Section "High Contrast Mode" in Article "Operating System and Browser Accessibility Display Modes": https://a11yproject.com/posts/operating-system-and-browser-accessibility-display-modes/#high-contrast-mode

> When activated, it maximizes the contrast between OS content and the background 

That's not true. The HCM overwrites the colours, but it is also possible to select specific colours that have little contrast, e.g. if someone is sensitive to glare

> it is placed on top of, even at the expense of the appearance of the UI. 

That's not true. Only in an application that is not WCAG compliant can the appearance suffer. In other words, with all methods of adjusting the contrast, the appearance may suffer.

Much more crucial: In contrast to other methods of contrast adjustment, the HCM is the only method where the colors are not only adjusted but completely overwritten. However, this makes the HCM the only method that is compliant with the UAAG. See Guidlines 1.3 and 1.4 (https://www.w3.org/TR/UAAG20/)

> High Contrast Mode support is a little bit complicated. It is only available for Windows

As far as I know there are several GUI for Linux that also offer HCM

> For web content, High Contrast Mode will only work with Internet Explorer (versions 10 and 11) and Edge (any version) for Windows 7, 8, and 8.1

The HCM also works in Firefox

> For Windows 10, High Contrast Mode will work with web content for any browser.

This is not true: The HCM does not work in Chrome (so this browser is not compliant with UAAG)

>High Contrast Mode will automatically be applied to web content described semantically in HTML. For example, paragraphs of text described using the p element will automatically update when High Contrast Mode is activated

This is slightly misleading as it sounds like there is a separate color for all or many colors in the HCM, e.g. for the p-element. But this is not so. Only the following colors can be defined:
* background
* foreground (text)
* links
* buttons and form elements 
* disabled elements
* selected elements

**Comment by ericwbailey** [comment]
Thank you for your feedback, @JAWS-test. Some replies:

> That's not true. The HCM overwrites the colours, but it is also possible to select specific colours that have little contrast, e.g. if someone is sensitive to glare

I'm speaking more towards intent and less towards mechanism. Additionally, while possible, I would consider selecting specific colors an antipattern as you want to be relying on color keywords because HCM themes can be customized.

> That's not true. Only in an application that is not WCAG compliant can the appearance suffer. In other words, with all methods of adjusting the contrast, the appearance may suffer.

I am familiar with UAAG . An application can be WCAG compliant, but still not have sufficient contrast needs for a low vision user. Again, this is where HCM's ability to customize its theme colors kicks in.

> As far as I know there are several GUI for Linux that also offer HCM

If you can link me to some, I will happily add them to the post.

> The HCM also works in Firefox

I believe Firefox has incomplete support, and therefore I considered it not good to recommend as a browser, as the end user experience would suffer.

> This is not true: The HCM does not work in Chrome (so this browser is not compliant with UAAG)

My understanding is [this work has been completed](https://twitter.com/soMelanieSaid/status/1217197651906613248).

> This is slightly misleading as it sounds like there is a separate color for all or many colors in the HCM, e.g. for the p-element. But this is not so. 

I am aware of this as well, but am trying to phrase the article in a way that is approachable. Our [Content Style Guide](https://github.com/a11yproject/a11yproject.com/blob/gh-pages/CONTENT_STYLE_GUIDE.md#reading-level) discusses this, specifically the Themes and Reading Level sections. 

If you have a suggestion for an alternate phrasing that can encapsulate your concerns while honoring this, I'll happily update it.

**Comment by JAWS-test** [comment]
@ericwbailey 

> I'm speaking more towards intent and less towards mechanism.

Intent of the HCM or the users who use HCM? The intent of the users can be to use lower contrasts or simply to exchange light and dark (i.e. foreground and backround, without improving the contrast)

> Additionally, while possible, I would consider selecting specific colors an antipattern

This is not an antipattern, but exactly the big advantage of HCM in opposition to other tools that only adjust the contrast (like Chrome). And it is exactly what the UAAG demands. Users need specific colors and can select them in the HCM.

> I believe Firefox has incomplete support, and therefore I considered it not good to recommend as a browser, as the end user experience would suffer

The Firefox support is very good. There is only one deviation: The colors of the links are defined by the user in Firefox itself and are not taken over by the operating system. But this only means that the user has to adapt the link colors to his needs in Firefox once and can then work with HCM as usual.

> My understanding is this work has been completed.

The fact that Microsoft Edge supports HCM (and even better than Internet Explorer) does not mean that Chrome supports it, at least not to my knowledge. In the current Chrome (version 80) HCM does not work.

> I am aware of this as well, but am trying to phrase the article in a way that is approachable. Our Content Style Guide discusses this, specifically the Themes and Reading Level sections

I am not suggesting to make the text more complicated, but simpler. Instead of the p-element I would only write that in the HCM the background, all text content, all links and all form elements (incl. buttons) get the color defined by the user

**Comment by JAWS-test** [comment]
> In situations where the HTML cannot be modified, High Contrast Mode styling can be tweaked. For Internet Explorer versions 10 and 11, you can use the the -ms-high-contrast media feature in CSS. In these situations, you should use a specialized subset of CSS2 system color keywords, as High Contrast Mode theme colors are dynamic.

I suggest that this paragraph be removed or thoroughly revised. Removal would be preferable:
* Internet Explorer is an outdated and rarely used browser
* The paragraph is difficult to understand
* The paragraph is wrong


If the paragraph is revised, please note that -ms-high-contrast and the Windows system colors have nothing to do with each other.

* -ms-high-contrast is used to determine whether HCM is on. If HCM is on, background graphics can be replaced by text, for example. This is only possible in IE 11. The workaround for Firefox is to use JS to determine whether background graphics are displayed.
* Windows system colors are deprecated and should no longer be used, but work in IE11, Firefox and Chrome.  However, they only work if no HCM is on. With HCM on they behave like all other colors and are replaced by the corresponding color defined in the HCM. The Windows system colors can be used sensibly within SVG graphics because they are then replaced in the HCM with the corresponding color values.

**Comment by JAWS-test** [comment]
> Edge does not fully support -ms-high-contrast. Instead, new Chromium-based versions of Edge support the forced-colors media query. Because of this, it is important to test using both Internet Explorer and Edge to make sure your High Contrast Mode styling work looks the way it is intended to.

This is also not true. Chromium Edge fully support -ms-high-contrast.

forced-colors media query
* is not supported by IE 11, Chrome and Firefox. 
* is supported by Chromium Edge

prefers-color-scheme
* is supported by Chome for HCM and Dark Mode
* is supported by Firefox only for Dark Mode
* is supported by Chromium Edge only for HCM

**Comment by ericwbailey** [comment]
> This is not an antipattern, but exactly the big advantage of HCM in opposition to other tools that only adjust the contrast (like Chrome). And it is exactly what the UAAG demands. Users need specific colors and can select them in the HCM.

The advice is for _devlopers_ to not pick exact colors, so as to facilitate _users_ being able to achieve this.

> The Firefox support is very good. There is only one deviation: The colors of the links are defined by the user in Firefox itself and are not taken over by the operating system. But this only means that the user has to adapt the link colors to his needs in Firefox once and can then work with HCM as usual.

If there is not perfect support then I do not feel comfortable recommending it for use, as it will have inconsistent behavior. And I say this as a loyal Firefox user.

> The fact that Microsoft Edge supports HCM (and even better than Internet Explorer) does not mean that Chrome supports it, at least not to my knowledge. In the current Chrome (version 80) HCM does not work.

I will remove this, but re-enter the advice when Chrome is compliant. I will also try to update your subsequent comment.

> I suggest that this paragraph be removed or thoroughly revised. Removal would be preferable:

IE still enjoys a lot of use, especially in the corporate sector. CSS2 color keywords are the only reliable way to ensure support for dynamically updating HCM colors. This section will not change.

**Comment by JAWS-test** [comment]
> The advice is for devlopers to not pick exact colors, so as to facilitate users being able to achieve this.

No, I don't understand that. Do you think that no colors should be defined at all to comply with WCAG 1.4.8? This is not necessary with HCM, because the colors are automatically overwritten, whether they are explicitly defined by the developer or not.

> CSS2 color keywords are the only reliable way to ensure support for dynamically updating HCM colors

CSS2 color keywords are overwritten by HCM just like all other colors (except inside SVG). See https://codepen.io/jaws-test/pen/abOQNOP

**Comment by ericwbailey** [comment]
Color is user defined in selecting a HCM theme. The color will default to the OS' HCM mechanism's default theme, unless another HCM theme is chosen, or a user specifies a custom HCM theme.

**Comment by ericwbailey** [comment]
Closed via https://github.com/a11yproject/a11yproject.com/commit/dd5263eac78d1bd7018e14e6e8ed264a83e221c3.

@JAWS-test This exchange was not pleasant for me as a site maintainer. You did not provide resources when asked, conflated opinion with fact, and misrepresented a lot of information, forcing us to re-research material referenced to write this article. In addition, your tone made this dialog unnecessarily tense.

This is an open source project, where the volunteers make commitments of their own time to keep the site running. I would suggest you consider these sorts of things when constructively and courteously interacting with other open source projects in the future. As for The A11Y Project, please do not raise issues here again.
